Sources/StreamVideo/Utils/AudioRecorder/StreamAudioRecorder.swift

```diff
@@ -0,0 +1,180 @@
+//
+// Copyright © 2024 Stream.io Inc. All rights reserved.
+//
+
+import AVFoundation
+import Combine
+import Foundation
+
+/// This class abstracts the usage of AVAudioRecorder, providing a convenient way to record and manage
+/// audio streams. It handles setting up the recording environment, starting and stopping recording, and
+/// publishing the average power of the audio signal. Additionally, it adjusts its behavior based on the
+/// presence of an active call, automatically stopping recording if needed.
+open class StreamAudioRecorder: @unchecked Sendable {
```

Comment:
Do you think this might collide with the voice recording feature from chat?

Reply:
It does :/ Given that they are in different frameworks they will still be accessed correctly. Just for convenience would you prefer renaming it to `StreamCallAudioRecorder`?

Reply:
I'm not worried about the naming - the namespaces will handle that. More about the usage - e.g. if you send a voice message on the call, or even before/after.

Reply:
It shouldn't have any different effect than it does now. As it seems we don't have async messages on SwiftUI chat and i can't test it. I believe that a custom handling for the AudioSession will be required in order when an async message gets sent we don't disable the AudioSession.

Reply:
We have async messages on SwiftUI - they are probably not turned on (`utils.composerConfig.isVoiceRecordingEnabled`). Might be good to test it.

Reply:
Just tested it. In order to enable async messages on ChatSwiftUI we need to provide a custom AudioSessionConfiguration (like we do on UIKit). That requires an update on ChatSwiftUI. Until then we cannot async messages on the demo app.